src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,496 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import re
+from collections import defaultdict
+from functools import cached_property
+from typing import Optional, Union, List, TYPE_CHECKING
+
+from snowflake.snowpark._internal.data_source import DataSourcePartitioner
+from snowflake.snowpark._internal.data_source.drivers import BaseDriver
+from snowflake.snowpark._internal.utils import (
+    generate_random_alphanumeric,
+    random_name_for_temp_object,
+    TempObjectType,
+)
+from snowflake.snowpark.types import (
+    StructType,
+    DecimalType,
+    FloatType,
+    StringType,
+    DateType,
+    BooleanType,
+    ArrayType,
+    StructField,
+    VariantType,
+    TimestampType,
+    TimestampTimeZone,
+)
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session
+    from snowflake.snowpark.dataframe import DataFrame
+
+logger = logging.getLogger(__name__)
+
+# reference for java type to snowflake type:
+# https://docs.snowflake.com/en/developer-guide/udf-stored-procedure-data-type-mapping
+
+JAVA_TYPE_TO_SNOWFLAKE_TYPE = {
+    "java.math.BigDecimal": DecimalType,
+    "java.math.BigInteger": DecimalType,
+    "java.lang.Float": FloatType,
+    "java.lang.Double": FloatType,
+    "java.lang.Boolean": BooleanType,
+    "java.lang.String": StringType,
+    "java.sql.Timestamp": TimestampType,
+    "java.sql.Date": DateType,
+    "java.sql.Array": ArrayType,
+    "java.lang.String[]": ArrayType,
+    "java.lang.Long": DecimalType,
+    "java.lang.Integer": DecimalType,
+    "java.lang.Short": DecimalType,
+}
+JDBC_TYPE_TO_SNOWFLAKE_TYPE = {
+    "NUMERIC": DecimalType,
+    "VARCHAR": StringType,
+    "CHAR": StringType,
+    "CLOB": StringType,
+    "NCHAR": StringType,
+    "NVARCHAR": StringType,
+    "NCLOB": StringType,
+    "TIMESTAMP": TimestampType,
+    "VARBINARY": StringType,
+}
```

Comment:
I see there are more types in JDBC spec like double/float/bool/time, ref: https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html, we can't support them?

we are not supporting them?